### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by scolladon**
## Description

Upgrade every dependencies

```sh
$ ncu -u
$ npm install
```

Migrate to husky v5 (breaking change) => requires `npm install`

## Motivation and Context

Having update dev dependencies
Having a secured up to date application
Benefit from the new features of the tools we relayed on

## How Has This Been Tested?

```sh
npm run test
```

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,33 +18,30 @@
     "test:unit:coverage": "sfdx-lwc-jest --coverage -- --runInBand",
     "prettier": "prettier --write \"**/*.{cls,cmp,component,css,html,js,json,md,page,trigger,xml,yaml,yml}\"",
     "prettier:verify": "prettier --list-different \"**/*.{cls,cmp,component,css,html,js,json,md,page,trigger,xml,yaml,yml}\"",
-    "build:playground": "rollup --config ./rollup/rollup.config.js"
+    "build:playground": "rollup --config ./rollup/rollup.config.js",
+    "pre-commit": "lint-staged && npm run lint",
+    "pre-push": "npm run test:coverage",
+    "prepare": "husky install"
```
Comment (victorgz) [inline package.json:24]:
```suggestion
    "pre-push": "npm run test:coverage"
```

File: sfdx-project.json [hunk 2]
```diff
@@ -7,5 +7,5 @@
   ],
   "namespace": "",
   "sfdcLoginUrl": "https://login.salesforce.com",
-  "sourceApiVersion": "50.0"
+  "sourceApiVersion": "51.0"
```
Comment (victorgz) [inline sfdx-project.json:10]: @scolladon maybe we can also upgrade the API version of the other metadata files, like the components? What do you think?
Suggestion: we can use this plugin https://github.com/muenzpraeger/sfdx-plugin with the command `sfdx mzpr:source:api:set`... and let the magic happens ! 😄

Comment (scolladon) [reply sfdx-project.json:10]: That is a very good idea !
I add it in this PR

File: package.json (hunk 1 above)
Comment (scolladon) [reply package.json:24]: What is bothering you here @victorgz ?

Comment (victorgz) [reply package.json:24]: @scolladon arg forget about it !!! it was just a lapsus, I read by mistake in my head: "prepare" : "echo husky install", and I thought it was something you had used for testing 😃

**Review by victorgz — APPROVED** [review]

**Pull request merged**